src/compiler/checker.ts

```diff
@@ -9580,6 +9580,10 @@ namespace ts {
                 return Ternary.False;
             }
 
+            function isIgnoredJsxProperty(source: Type, sourceProp: Symbol, targetMemberType: Type | undefined) {
+                return source.flags & TypeFlags.JsxAttributes && !(isUnhyphenatedJsxName(sourceProp.escapedName) || !!targetMemberType);

```

Comment:
3 usages of `isUnhyphenatedJsxName` are not negated, while 2 usages of `isUnhyphenatedJsxName` are. While the negation of the negation is linguistically unfortunate at the use site, I think the name is the most straightforward and self-documenting it can be as-is.